Arriba/Arriba.Web/parts/Grid.jsx

```diff
@@ -330,34 +320,21 @@ export default class Grid extends EventedComponent {
     }
     componentDidMount() {
         super.componentDidMount();
-        var defaultQuery = (this.props.params.p === "default" ? configuration.gridDefault : this.props.params.p);
-        if (defaultQuery) {
+        var defaultQuery = this.props.params.p || configuration.gridDefault;
+        if (!this.props.query && defaultQuery) {
             this.selectDefaultQuery(defaultQuery);
         }
         this.componentDidUpdate({}, {});
     }
     componentDidUpdate(prevProps, prevState) {
+        const diffProps = Object.diff(prevProps, this.props);
         const diffState = Object.diff(prevState, this.state);
 
-        if (diffState.hasAny("query")) {
-            // Only query every 250 milliseconds while typing
-            this.timer = this.timer || window.setTimeout(() => this.setState({ debouncedQuery: this.state.query }), 250);
-        }
-
-        if (diffState.hasAny("debouncedQuery")) {
-            this.getCounts();
-        }
-
-        if (diffState.hasAny("userSelectedTable", "counts")) {
-            const currentTable = this.state.userSelectedTable || this.state.counts && this.state.counts.resultsPerTable[0].tableName;
-            this.setState({ currentTable: currentTable });
-        }
-
-        if (diffState.hasAny("currentTable") && prevState.currentTable) {
+        if (diffProps.hasAny("currentTable") && prevProps.currentTable) {
             this.setState(this.getClearedUserSelections());
         }
 
-        if (diffState.hasAny("debouncedQuery", "currentTable", "aggregationFunction", "aggregateColumn", "rows", "rowLabels", "cols", "colLabels", "show", "showPortionOf", "showPortionAs")) {
+        if (diffProps.hasAny("debouncedQuery", "currentTable") || diffState.hasAny("aggregationFunction", "aggregateColumn", "rows", "rowLabels", "cols", "colLabels", "show", "showPortionOf", "showPortionAs")) {
             this.getGrid();

```

Comment:
It would be cool (though I guess not strictly necessary) if we didn't re-query the service if we only see display changes. Labels, Show, ShowPortionOf, and ShowPortionAs don't require a re-query, just a re-render.

Reply:
Good point. I think I can just remove those three detections and it re-render will still happen.